src/test/java/org/takes/http/BkBasicTest.java

```diff
@@ -108,4 +179,91 @@ public void exec(final URI home) throws IOException {
             }
         );
     }
+
+    /**
+     * Try to send 2 requests in 1 tcp connection.
+     * @throws IOException Socket IOException.
+     */
+    @Test
+    public void persistenceConnection() throws IOException {
+        final Socket socket = Mockito.mock(Socket.class);
+        Mockito.when(socket.getInputStream()).thenReturn(
+            new ByteArrayInputStream(TWO_POST_REQUESTS)
+        );
+        Mockito.when(socket.getLocalAddress()).thenReturn(
+            InetAddress.getLocalHost()
+        );
+        Mockito.when(socket.getLocalPort()).thenReturn(0);
+        Mockito.when(socket.getInetAddress()).thenReturn(
+            InetAddress.getLocalHost()
+        );
+        Mockito.when(socket.getPort()).thenReturn(0);
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Mockito.when(socket.getOutputStream()).thenReturn(baos);
+        new BkBasic(new TkGreedy(new TkText(HELLO_WORLD))).accept(socket);
```

Comment:
@hs3180 I still not understand how you check that the socket still open after first request...